File: 547-number-of-provinces/number-of-provinces.py
class Solution(object):
    def findCircleNum(self, graph):
        """
        :type isConnected: List[List[int]]
        :rtype: int
        """
        visited=[0]*len(graph)
        vertices=len(graph)
        count=0
        for i in range(vertices):
            if visited[i]==0:
                count+=1
                self.dfs(i,visited,graph)
        return count
    def dfs(self,i,visited,graph):
        visited[i]=1
        for j in range(len(graph)):
            if visited[j]==0 and graph[i][j] and i!=j:
                self.dfs(j,visited,graph)
    

        

File: 547-number-of-provinces/number-of-provinces.cpp
class Solution {
public:
    void dfs(vector<int>& visited,vector<vector<int>>& isConnected,int i){
        int n=visited.size();
        for (int j=0;j<n;j++){
            if (visited[j]==0 && isConnected[i][j]){
                    visited[j]=1;
                    dfs(visited,isConnected,j);
            }
        }
    }
    int findCircleNum(vector<vector<int>>& isConnected) {
        int n=isConnected.size();
        vector<int> visited(n,0);
        int ncomps=0;
        for (int i=0;i<n;i++){
            if (visited[i]==0){
                visited[i]=1;
                ncomps+=1;
                dfs(visited,isConnected,i);
            }
        }
        
        return ncomps;
    }
};